llrt_modules/src/modules/process.rs

```diff
@@ -14,3 +23,152 @@ pub fn get_platform() -> &'static str {
         _ => platform,
     }
 }
+
+fn cwd() -> String {
```

Comment:
This seems like it would cause a lifetime error, or is there some other way to return a &str?

```
fn cwd() -> &str {
     env::current_dir().unwrap().to_string_lossy().as_ref()
}
```

```
error[E0515]: cannot return value referencing temporary value
  --> llrt_modules/src/modules/process.rs:28:6
   |
28 |      env::current_dir().unwrap().to_string_lossy().as_ref()
   |      ---------------------------------------------^^^^^^^^^
   |      |
   |      returns a value referencing data owned by the current function
   |      temporary value created here
```

Reply:
Yeah you can avoid the double alloc by returning a `String<'js>` built from the `&str`

Reply:
Am I understanding this correctly?
```
fn cwd(ctx: Ctx<'_>) -> std::io::Result<Result<rquickjs::String<'_>>> {
    env::current_dir().map(|s| rquickjs::String::from_str(ctx, s.to_string_lossy().as_ref()))
}
```

Reply:
use an `and_then` to avoid nesting your result, otherwise yes

Reply:
Where do I add and_then? and_then needs to wrap with Ok, perhaps using current_dir().unwrap(), which we all know almost never fails.
```
env::current_dir().and_then(|path|Ok(rquickjs::String::from_str(ctx, path.to_string_lossy().as_ref())))
```

fmt
```
env::current_dir().and_then(|path| {
    Ok(rquickjs::String::from_str(
        ctx,
        path.to_string_lossy().as_ref(),
    ))
})
```

Reply:
```rust
fn cwd<'js>(ctx: Ctx<'js>) -> Result<rquickjs::String<'js>> {
    let cwd = env::current_dir().or_throw(&ctx)?;
    let cwd = cwd.to_string_lossy();
    let cwd = cwd.as_ref();
    rquickjs::String::from_str(ctx, cwd)
}
```